### PR TITLE
update kube and dependencies to v1.18.0.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal hchiramm@redhat.com

```release-note
Action required: client-go gets updated to Kubernetes 1.18.0 which changes the API of several functions (context added, create/delete/update options).
```

### CODE REVIEW

**Comment by humblec** [comment]
@pohly Please review.

**Comment by msau42** [comment]
@humblec can you add a release note?

**Comment by humblec** [comment]
> @humblec can you add a release note?

@msau42 done.. ptal.

**Comment by pohly** [comment]
> done.. ptal.

This is very verbose. Something shorter like "Action required: client-go gets updated to Kubernetes 1.18.0 which changes the API of several functions (context added, create/delete/update options)" should be enough.

According to the discussion in https://kubernetes.slack.com/archives/C5P3FE08M/p1586337032010800, this is a major API change for libraries because the updated csi-lib-utils isn't guaranteed to be usable in downstream components unless those components adapt to the modified client-go interface.

We can merge this PR and bump the /v suffix of the module as part of preparing the next release.

**Comment by humblec** [comment]
> > done.. ptal.
> 
> This is very verbose. Something shorter like "Action required: client-go gets updated to Kubernetes 1.18.0 which changes the API of several functions (context added, create/delete/update options)" should be enough.
> 
> According to the discussion in https://kubernetes.slack.com/archives/C5P3FE08M/p1586337032010800, this is a major API change for libraries because the updated csi-lib-utils isn't guaranteed to be usable in downstream components unless those components adapt to the modified client-go interface.
> 
> We can merge this PR and bump the /v suffix of the module as part of preparing the next release.

@pohly completely accepted the release note as you suggested :). PTAL

**Comment by humblec** [comment]
@msau42 ptal.

File: go.mod [hunk 1]
```diff
@@ -3,69 +3,68 @@ module github.com/kubernetes-csi/csi-lib-utils
...
+	github.com/googleapis/gnostic v0.3.1 // indirect
+	github.com/onsi/ginkgo v1.12.0 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	github.com/stretchr/testify v1.5.1
+	golang.org/x/net v0.0.0-20200324143707-d3edc9973b7e
+	golang.org/x/sys v0.0.0-20200406155108-e3b113bbe6a4 // indirect
+	google.golang.org/genproto v0.0.0-20200406120821-33397c535dc2 // indirect
+	google.golang.org/grpc v1.28.0
+	k8s.io/api v0.18.0
+	k8s.io/client-go v1.5.1
```
Comment (msau42) [inline go.mod:19]: Is this the correct version of client-go?

Comment (humblec) [reply go.mod]: ah.. corrected it @msau42

**Comment by humblec** [comment]
@msau42 just wondering how the `lgtm` , `approved` label vanished from this PR .. :)

**Pull request merged**